proposals/first-class-span-types.md

```diff
@@ -121,6 +121,55 @@ The compiler expects to use the following helpers or equivalents to implement th
 | ReadOnlySpan to ReadOnlySpan | `static ReadOnlySpan<T>.CastUp<TDerived>(ReadOnlySpan<TDerived>)` |
 | string to ReadOnlySpan | `static ReadOnlySpan<char> MemoryExtensions.AsSpan(string)` |
 
+#### Overload resolution
+
+*Better conversion from expression* ([§12.6.4.5][better-conversion-from-expression]) is updated to prefer implicit span conversions.
+This is based on [collection expressions overload resolution changes][ce-or].
+
+> Given an implicit conversion `C₁` that converts from an expression `E` to a type `T₁`, and an implicit conversion `C₂` that converts from an expression `E` to a type `T₂`, `C₁` is a *better conversion* than `C₂` if one of the following holds:
+>
+> - `E` is a *collection expression* and one of the following holds:
+>   - `T₁` is `System.ReadOnlySpan<E₁>`, and `T₂` is `System.Span<E₂>`, and an implicit conversion exists from `E₁` to `E₂`.
+>   - `T₁` is `System.ReadOnlySpan<E₁>` or `System.Span<E₁>`, and `T₂` is an *array_or_array_interface* with *element type* `E₂`, and an implicit conversion exists from `E₁` to `E₂`.
+>   - `T₁` is not a *span_type*, and `T₂` is not a *span_type*, and an implicit conversion exists from `T₁` to `T₂`.
+> - `E` is not a *collection expression* and one of the following holds:
+>   - `E` exactly matches `T₁` and `E` does not exactly match `T₂`
+>   - `E` exactly matches both or neither of `T₁` and `T₂`, and `T₁` is a better conversion target than `T₂`
```

Comment:
>- `E` exactly matches both or neither of `T₁` and `T₂`, and `T₁` is a better conversion target than `T₂` [](http://example.com/codeflow?start=4&length=104)

In the call to `M(new string[0])`, will we prefer `M(object[])` over `M(ReadOnlySpan<object>)`? If so, is that intended?

Reply:
According to the spec here we would prefer `object[]` since it's implicitly convertible to `ROS<object>` (and not the other way around) hence it's a better conversion target.

The intention was to prefer implicit span conversion, so this goes against that intention. I will update the rule. Thanks.

I'm not sure if in general it makes sense for `ROS<object>` to be the better target for `string[]` than `object[]`. I guess one potential advantage of ROS is that it's readonly, unlike `object[]` which the method could potentially assign a non-string object into?